tests/phpunit/Logger/LoggerFactoryTest.php

```diff
@@ -0,0 +1,184 @@
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Logger;
+
+use Generator;
+use Infection\Configuration\Entry\Badge;
+use Infection\Configuration\Entry\Logs;
+use Infection\Console\LogVerbosity;
+use Infection\Logger\BadgeLogger;
+use Infection\Logger\DebugFileLogger;
+use Infection\Logger\LoggerFactory;
+use Infection\Logger\PerMutatorLogger;
+use Infection\Logger\SummaryFileLogger;
+use Infection\Logger\TextFileLogger;
+use Infection\Mutant\MetricsCalculator;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+final class LoggerFactoryTest extends TestCase
+{
+    public function test_it_creates_no_logger_for_low_verbosity(): void
+    {
+        $factory = new LoggerFactory(
+            new MetricsCalculator(),
+            new Filesystem(),
+            LogVerbosity::NONE,
+            true,
+            true
+        );
+
+        $loggers = $factory->createLoggersFromLogEntries(
+            new Logs(
+                '/a/file',
+                '/a/file',
+                '/a/file',
+                '/a/file',
+                null
+            ),
+            $this->createMock(OutputInterface::class)
+        );
+
+        $this->assertCount(0, $loggers);
+    }
+
+    /**
+     * @dataProvider provideLogTypesAndClasses
+     */
+    public function test_it_creates_logger_for_log_type(Logs $logs, string $expectedLogClass): void
+    {
+        $factory = new LoggerFactory(
+            new MetricsCalculator(),
+            new Filesystem(),
+            LogVerbosity::NORMAL,
```

Comment:
if the log verbosity may play a role (which is what `test_it_creates_a_bade_logger_on_low_verbosity` hints at), then it may be worth adding it as an input